pkg/services/ngalert/remote/client/alertmanager.go

```diff
@@ -24,36 +26,40 @@ type AlertmanagerConfig struct {
 
 type Alertmanager struct {
 	*amclient.AlertmanagerAPI
-	httpClient *http.Client
+	httpClient client.Requester
 	url        *url.URL
 	logger     log.Logger
 }
 
-func NewAlertmanager(cfg *AlertmanagerConfig) (*Alertmanager, error) {
+func NewAlertmanager(cfg *AlertmanagerConfig, metrics *metrics.RemoteAlertmanager) (*Alertmanager, error) {
 	// First, add the authentication middleware.
 	c := &http.Client{Transport: &MimirAuthRoundTripper{
 		TenantID: cfg.TenantID,
 		Password: cfg.Password,
 		Next:     http.DefaultTransport,
 	}}
 
+	tc := client.NewTimedClient(c, metrics.HTTPRequestDuration)
 	apiEndpoint := *cfg.URL
 
 	// Next, make sure you set the right path.
 	u := apiEndpoint.JoinPath(alertmanagerAPIMountPath, amclient.DefaultBasePath)
-	transport := httptransport.NewWithClient(u.Host, u.Path, []string{u.Scheme}, c)
+
+	// Create an Alertmanager client using the timed client as the transport.
+	r := httptransport.New(u.Host, u.Path, []string{u.Scheme})
+	r.Transport = tc
 
 	return &Alertmanager{
 		logger:          cfg.Logger,
 		url:             cfg.URL,
-		AlertmanagerAPI: amclient.New(transport, nil),
-		httpClient:      c,
+		AlertmanagerAPI: amclient.New(r, nil),
+		httpClient:      tc,
```

Comment:
Should this be renamed to requester now?

Reply:
This is the client we use to make HTTP requests that are not possible with the upstream Alertmanager client (e.g. readiness check).
I'd like to keep the name, it does not have to do with the struct/interface behind it, but with the fact that it's used to do plain HTTP requests. Wdyt?